internal/pkg/agent/application/monitoring/process.go

```diff
@@ -55,16 +57,26 @@ func processHandler(coord *coordinator.Coordinator, statsHandler func(http.Respo
 			return statsHandler(w, r)
 		}
 
-		metricsPath := vars[metricsPathKey]
-		if _, ok := redirectPathAllowlist[metricsPath]; ok {
-			if isProcessRedirectable(componentID) {
-				return redirectToPath(w, r, componentID, metricsPath, operatingSystem)
+		if isProcessRedirectable(componentID) {
+			// special handling for redirectable processes
+			// apm needs its own output even for no path
+			metricsPath := vars[metricsPathKey]
+			_, ok := redirectPathAllowlist[metricsPath]
+			if !ok {
+				return errorfWithStatus(http.StatusNotFound, "process specified does not expose metrics")
 			}
-			return errorfWithStatus(http.StatusNotFound, "process specified does not expose metrics")
-		} else if strings.HasPrefix(componentID, fleetServerPrefix) {
-			// special case, fleet server is expected to return stats right away
-			// removing this would be breaking
-			return redirectToPath(w, r, componentID, "stats", operatingSystem)
+
+			if strings.HasPrefix(componentID, fleetServerPrefix) && metricsPathKey == "" {
+				// special case, fleet server is expected to return stats right away
+				// removing this would be breaking
+				metricsPath = "stats"
+			}
+			if strings.HasPrefix(componentID, apmPrefix) {
+				// from binary name back to input type, keep the output name as is (apm-default)
+				componentID = strings.Replace(componentID, apmPrefix, apmTypePrefix, -1)
```

Comment:
Is this covered in the unit tests? My biggest concern with all this special handling is that we will accidentally break it again because we have no test coverage for it.